### ISSUE TITLE
a11y: terms show/hide should be a button not a link

### ISSUE THREAD

**Issue by ramiy** [WordImpress/Give#1177]
## Issue Overview

The current "Show Term" is not accessible. We need to rewrite the code. To add "role" tags, to add "aria" tags to change the HTML tags.

![terms-a11y](https://cloud.githubusercontent.com/assets/576623/19837668/7c9e7cec-9ec7-11e6-93ae-34f8483c3f67.png)
## Todos
- [x] replace `<a>` with  `<button>` or with `<a role="button">`.
- [x] use `aria-controls` on the button to point to the terms DIV.
- [ ] use `aria-expanded` on the button to tell the screen reader if the terms are hidden/displayed.
- [ ] use `aria-hidden` on the terms text DIV to tell the screen reader that the terms are hidden/displayed.

**Comment by ramiy** [comment]
If you would like to know more about this issue you can find many tutorials and examples around the web but I think [this example is the best](http://oaa-accessibility.org/example/22/), it has all the components together.

**Comment by ramiy** [comment]
### `<button>` or `<a role="button">` ?

Using `<button>`:

![terms-button](https://cloud.githubusercontent.com/assets/576623/19840122/f64dbf96-9ef7-11e6-92de-a1c6e3a248d7.png)

When changing to `<button>` we inherit the style, this is why I used `role="button"`.

**Comment by DevinWalker** [comment]
@ramiy let's use `role="button"`

**Comment by ravinderk** [comment]
@DevinWalker @ramiy @mathetos  I notice that focus will be auto set when we will use the tab on `Show Terms` link but after clicking this link, I think the focus should be transfer to `Hide Term` link so that user we will continue interacting with keyboard also.

Let me know your views on that.

![mrgfbqtfz5](https://cloud.githubusercontent.com/assets/1784821/20350316/701b8cba-ac34-11e6-93b1-0ac427484dc0.gif)

**Comment by mathetos** [comment]
@ravinderk Good catch. Maintaining good keyboard navigation sounds great, particularly since that sounds like an easy fix.

**Comment by DevinWalker** [comment]
@ravinderk when you tab from the email field you're saying the focus should be on the checkbox, right?

**Comment by ravinderk** [comment]
@DevinWalker I am not talking about that focus should be on `Agree to terms` checkbox after email field. I am saying that after clicking `Show Terms` focus should be on `Hide Terms` button instead of any. Currently, I have click tab one more time to set focus to `Hide Terms` but i think it should be automatic after clicking on `Show Terms`.
